guides/source/upgrading_ruby_on_rails.md

```diff
@@ -272,6 +272,78 @@ You can invalidate the cache either by touching the product, or changing the cac
 <% end %>
 ```
 
+#### Migrating from Marginalia
+
+Active Record ships with the same default configuration as Marginalia. For basic use, enable the feature in your application configuration:
+
+```ruby
+# config/application.rb
+module MyApp
+  class Application < Rails::Application
+    config.active_record.query_log_tags_enabled = true
+  end
+end
+```
+
+For a more involved configuration query log tags operate in a similar way to Marginalia, but instead of defining new methods to be called at runtime you pass Procs or static values to the configuration:
+
+```ruby
+# config/application.rb
+module MyApp
+  class Application < Rails::Application
+    config.active_record.query_log_tags = [
+      :application,
+      {
+        dynamic_tag: -> { "Dynamic tag #{Time.now}" },
+        static_tag: "value"
+      }
+    ]
+  end
+end
+```
+
+Instead of referring to `marginalia_controller`, any object can be stored in the `context` and then referenced in a tag definition. The default configuration will set the context with `controller` & `job` as appropriate.
+
+```ruby
+class MyController < ApplicationController
+  def controller_info
```

Comment:
Can't have this as a public instance method, that'll expose it as an action. Can we point to other documentation instead that shows what's exposed in the context and when?